plugins/BEdita/API/src/Controller/ResourcesController.php

```diff
@@ -97,15 +94,126 @@ protected function findAssociation($relationship)
     {
         $relationship = Inflector::underscore($relationship);
         if (array_key_exists($relationship, $this->getConfig('allowedAssociations'))) {
-            $associations = TableRegistry::get($this->modelClass)->associations();
+            $associations = $this->Table->associations();
             foreach ($associations as $association) {
                 if ($association->property() === $relationship) {
                     return $association;
                 }
             }
         }
 
-        throw new NotFoundException(__('Relationship "{0}" does not exist', $relationship));
+        throw new NotFoundException(__d('bedita', 'Relationship "{0}" does not exist', $relationship));
+    }
+
+    /**
+     * Paginated list of resources.
```

Comment:
The problem with the previous approach was that if a request with a not supported HTTP verb reached a known endpoint, the response would have been a "Not Found" instead of a "Method Not Allowed" (i.e. `PUT /users` would have returned 404).

Indeed, this approach does break SRP, but allows us to correctly detect a method not allowed. Having protected methods `add()`, `edit()` and `delete()` and conditionally invoking them (`if ($this->request->is('post') { $this->add($entity); }`) is an option. However, we had already broken SRP in the `relationships()` method, that handles `GET`, `POST`, `PATCH` and `DELETE`, so we should probably refactor that as well.